perfkitbenchmarker/providers/gcp/gce_network.py

```diff
@@ -236,17 +246,29 @@ def __init__(self, network_spec):
                                                FLAGS.gce_subnet_addr,
                                                self.project)
     firewall_name = 'default-internal-%s' % FLAGS.run_uri
+    # allow 192.168.0.0/16 addresses
+    firewall_name2 = 'default-internal2-%s' % FLAGS.run_uri
+    # add support for cidr and separate networks
+    # @TODO alias RFC1918 private networks in a single rule
+    if network_spec.zone and network_spec.cidr:
+      firewall_name = 'default-internal-%s-%s' % (network_spec.zone, FLAGS.run_uri)
+      firewall_name2 = 'default-internal2-%s-%s' % (network_spec.zone, FLAGS.run_uri)
+      self.NETWORK_RANGE = network_spec.cidr
     self.default_firewall_rule = GceFirewallRule(
         firewall_name, self.project, ALLOW_ALL, name, NETWORK_RANGE)
+    self.default_firewall_rule2 = GceFirewallRule(
+        firewall_name2, self.project, ALLOW_ALL, name, NETWORK_RANGE2)
 
   @staticmethod
   def _GetNetworkSpecFromVm(vm):
     """Returns a BaseNetworkSpec created from VM attributes."""
-    return GceNetworkSpec(project=vm.project, zone=vm.zone)
+    return GceNetworkSpec(project=vm.project, zone=vm.zone, cidr=vm.cidr)
 
   @classmethod
   def _GetKeyFromNetworkSpec(cls, spec):
     """Returns a key used to register Network instances."""
+    if spec.zone and spec.cidr:
```

Comment:
hmm, i am not sure about zone being part of the key.

this implementation won't work if i want to setup vpn between 2 networks in same zone.